numpy/core/tests/test_multiarray.py

```diff
@@ -4264,7 +4264,7 @@ def _clip_type(self, type_group, array_max,
 
                 x = (np.random.random(1000) * array_max).astype(dtype)
                 if inplace:
-                    x.clip(clip_min, clip_max, x)
+                    x.clip(clip_min, clip_max, x, casting='unsafe')
```

Comment:
Is this still necessary with the change from int to float that you made below?

Reply:
Yes, this still causes problems on signed / unsigned integer promotion

Reply:
Maybe add a comment...